### PR TITLE
Return EINVAL if both ATIM & ATIM_NOW, MTIM & MTIM_NOW are set in path_filestat_set_times

### PR DESCRIPTION
**Pull request by marmistrz**
For https://github.com/CraneStation/wasi-misc-tests/pull/19

### CODE REVIEW

File: src/hostcalls_impl/fs.rs [hunk 1]
```diff
@@ -843,6 +843,12 @@ pub(crate) fn path_filestat_set_times(
         fst_flags
     );
 
+    if fst_flags & host::__WASI_FILESTAT_SET_ATIM & host::__WASI_FILESTAT_SET_ATIM_NOW != 0
+        || fst_flags & host::__WASI_FILESTAT_SET_MTIM & host::__WASI_FILESTAT_SET_MTIM_NOW != 0
```
Comment (sunfishcode) [inline src/hostcalls_impl/fs.rs]: The logic here isn't what's intended; `fst_flags & host::__WASI_FILESTAT_SET_ATIM & host::__WASI_FILESTAT_SET_ATIM_NOW` always evaluates to zero.

Comment (marmistrz) [reply src/hostcalls_impl/fs.rs]: What guarantees that they will never be set together? Nothing prevents me from setting `__wasi_fstflags_t` to `host::__WASI_FILESTAT_SET_ATIM & host::__WASI_FILESTAT_SET_ATIM_NOW` and this is what one of the tests in wasi-misc-tests does (for the fd_* version): https://github.com/CraneStation/wasi-misc-tests/pull/14

Comment (sunfishcode) [reply src/hostcalls_impl/fs.rs]: `host::__WASI_FILESTAT_SET_ATIM & host::__WASI_FILESTAT_SET_ATIM_NOW` is zero. I agree that we need a check here, but we need the right check :-).

Comment (marmistrz) [reply src/hostcalls_impl/fs.rs]: :D you're right!

Comment (marmistrz) [reply src/hostcalls_impl/fs.rs]: fixed

**Comment by marmistrz** [comment]
I rebased and moved the check to the Unix-specific implementation, because it will be unnecessary for the Windows implementation proposed in #63.

**Comment by marmistrz** [comment]
Obsoleted by #64

**Pull request closed without merging**